platform/ui/src/components/SegmentationGroupTable/SegmentationGroupTable.tsx

```diff
@@ -31,71 +41,123 @@ const SegmentationGroupTable = ({
   setRenderInactiveSegmentations,
   setRenderOutline,
 }) => {
+  const [isConfigOpen, setIsConfigOpen] = useState(false);
+  const [activeSegmentationId, setActiveSegmentationId] = useState(null);
+
+  const onActiveSegmentationChange = segmentationId => {
+    onSegmentationClick(segmentationId);
+    setActiveSegmentationId(segmentationId);
+  };
+
+  useEffect(() => {
+    // find the first active segmentation to set
+    let activeSegmentationIdToSet = segmentations?.find(segmentation => segmentation.isActive)?.id;
+
+    // If there is no active segmentation, set the first one to be active
+    if (!activeSegmentationIdToSet && segmentations?.length > 0) {
+      activeSegmentationIdToSet = segmentations[0].id;
+    }
+
+    // If there is no segmentation, set the active segmentation to null
+    if (segmentations?.length === 0) {
+      activeSegmentationIdToSet = null;
+    }
+
+    setActiveSegmentationId(activeSegmentationIdToSet);
+  }, [segmentations, activeSegmentationId]);
```

Comment:
I don't think `activeSegmentationId` should be a dependency for this `useEffect`.